### DISCUSSION TITLE
How to run multiple requests in queue?

### DISCUSSION THREAD

**Discussion by robert-janaszek** [TanStack/query#3555] [Q&A]
I have long list of long-running queries and I want to run them in queue (eg 2 at the time). How can I achieve that in react-query? I can't run them all at once, because it will clog up the browser. I tried playing with `enabled` flag, but this causes massive amount of issues.

**Comment by TkDodo** [comment]
I think you'd need something like a request pooling library that sits between react-query and the network, something like [swimmer](https://github.com/tannerlinsley/swimmer) maybe?

**Comment by jclaveau** [comment]
There is a nice solution here using `async-mutex` https://github.com/TanStack/query/discussions/4943#discussioncomment-4925318
